library/Booster/Pattern/Base.hs

```diff
@@ -584,29 +588,24 @@ pattern KMap def keyVals rest <- Term _ (KMapF def keyVals rest)
                         ([], Nothing) -> mempty
                         ([], Just s) -> getAttributes s
                         (_ : _, Nothing) -> foldl1' (<>) $ concatMap (\(k, v) -> [getAttributes k, getAttributes v]) keyVals
-                        (_ : _, Just r) -> foldr (<>) (getAttributes r) $ concatMap (\(k, v) -> [getAttributes k, getAttributes v]) keyVals
+                        (_ : _, Just r) ->
+                            foldl' (<>) (getAttributes r) $ concatMap (\(k, v) -> [getAttributes k, getAttributes v]) keyVals
                 (keyVals', rest') = case rest of
-                    Just (KMap def' kvs r) | def' == def -> (kvs, r)
+                    Just (KMap def' kvs r) | def' == def -> (sortAndDeduplicate kvs, r)
```

Comment:
True. I've added this only because of [this](https://github.com/runtimeverification/hs-backend-booster/blob/960310e88a3d862c3c6ade854801f9138dcb89ad/unit-tests/Test/Booster/Pattern/InternalCollections.hs#L269) unit test that manually constructs a nested `KMap` with a broken invariant. This should never happen thought, so probably safe to remove the call to `sortAndDeduplicate` form here and also from the same case in `KSet`.